e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,141 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	"cosmossdk.io/math"
+	fiattokenfactorytypes "github.com/circlefin/noble-fiattokenfactory/x/fiattokenfactory/types"
+	"github.com/strangelove-ventures/interchaintest/v8"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFiatTFUpdateOwner(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	t.Parallel()
+
+	ctx := context.Background()
+
+	nw := nobleSpinUp(t, ctx, true)
+	noble := nw.chain
+	val := noble.Validators[0]
+
+	// ACTION: Update owner while TF is paused
+	// EXPECTED: Success; Pending owner set
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	w := interchaintest.GetAndFundTestUsers(t, ctx, "new-owner-1", math.OneInt(), noble)
+	newOwner1 := w[0]
+
+	_, err := val.ExecTx(ctx, nw.fiatTfRoles.Owner.KeyName(), "fiat-tokenfactory", "update-owner", newOwner1.FormattedAddress())
+	require.NoError(t, err, "error broadcasting update owner message")
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	// ACTION: Update owner from unprivileged account
+	// EXPECTED: Request fails; pending owner not set
+
+	w = interchaintest.GetAndFundTestUsers(t, ctx, "alice", math.OneInt(), noble)
+	alice := w[0]
+
+	_, err = val.ExecTx(ctx, alice.KeyName(), "fiat-tokenfactory", "update-owner", newOwner1.FormattedAddress())
+	require.ErrorContains(t, err, "you are not the owner: unauthorized")
+
+	// ACTION: Update Owner from blacklisted owner account
+	// EXPECTED: Success; pending owner set
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Owner)
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.Owner.KeyName(), "fiat-tokenfactory", "update-owner", newOwner1.FormattedAddress())
+	require.NoError(t, err, "error broadcasting update owner message")
+
+	// ACTION: Update Owner to a blacklisted account
+	// EXPECTED: Success; pending owner set
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, newOwner1)
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.Owner.KeyName(), "fiat-tokenfactory", "update-owner", newOwner1.FormattedAddress())
+	require.NoError(t, err, "error broadcasting update owner message")
+
+}
+
+func TestFiatTFAcceptOwner(t *testing.T) {
```

Comment:
I think there is a case missing here (and missing from the spec doc) about if the pending owner isn't specified (isn't set), then accepting the owner should fail.

Reply:
Added in https://github.com/noble-assets/noble/pull/367/commits/e03d7f4a64e0f240b56295936acb3e990a2a4242